modules/6_segmentation/src/segmentation.cpp

```diff
@@ -6,18 +6,60 @@ using namespace InferenceEngine;
 using namespace cv;
 using namespace cv::utils::fs;
 
+
 float Dice(const Mat& a, const Mat& b) {
-    CV_Error(Error::StsNotImplemented, "Dice score computation");
+    return 2.0f * countNonZero(a & b) / (countNonZero(a) + countNonZero(b));
 }
 
+
 ADAS::ADAS() {
     Core ie;
 
     // Load deep learning network into memory
     auto net = ie.ReadNetwork(join(DATA_FOLDER, "semantic-segmentation-adas-0001.xml"),
-                              join(DATA_FOLDER, "semantic-segmentation-adas-0001.bin"));
+        join(DATA_FOLDER, "semantic-segmentation-adas-0001.bin"));
+
+    InputInfo::Ptr inputInfo = net.getInputsInfo()["data"];
+
+    inputInfo->getPreProcess().setResizeAlgorithm(ResizeAlgorithm::RESIZE_BILINEAR);
+    inputInfo->setLayout(Layout::NHWC);
+    inputInfo->setPrecision(Precision::U8);
+
+    // Initialize runnable object on CPU device
+    ExecutableNetwork execNet = ie.LoadNetwork(net, "CPU");
+
+    // Create a single processing thread
+    req = execNet.CreateInferRequest();
+
+    outputName = net.getOutputsInfo().begin()->first;
 }
 
-void ADAS::segment(const Mat& image, Mat& mask) {
-    CV_Error(Error::StsNotImplemented, "ADAS semantic segmentation");
+
+static Blob::Ptr wrapMatToBlob(const Mat& m) {
+    CV_Assert(m.depth() == CV_8U);
+    std::vector<size_t> dims = { 1, (size_t)m.channels(), (size_t)m.rows, (size_t)m.cols };
+    return make_shared_blob<uint8_t>(TensorDesc(Precision::U8, dims, Layout::NHWC),
+        (uint8_t*)m.data);
 }
+
+
+void ADAS::segment(const Mat& image, Mat& mask) {
+    Blob::Ptr input = wrapMatToBlob(image);
+
+    req.SetBlob("data", input);
+
+    // Launch network
+    req.Infer();
+
+    // Copy output. 
+    int* output = req.GetBlob(outputName)->buffer();
+
+    for (int i = 0; i < req.GetBlob(outputName)->size(); i++)
+        mask.push_back(float(output[i]));
```

Comment:
Try to us more optimal [constructor](https://docs.opencv.org/master/d3/d63/classcv_1_1Mat.html#a51615ebf17a64c968df0bf49b4de6a3a)

Reply:
When I use the constructor, Mat is filled with some small floats instead of the normal numbers, although all dimensions are the same. Maybe I'm using it incorrectly.

-1st attempt: mask = Mat(req.GetBlob(outputName)->size(), 1, CV_32F, (float *)output);
-2nd attempt: mask = Mat(req.GetBlob(outputName)->size(), 1, CV_32F, output);
-3rd attempt: mask = Mat(req.GetBlob(outputName)->size(), 1, req.GetBlob(outputName)->size() * 1 * 1 * sizeof(float), output);

Reply:
Because you need to create `CV_32SC1` cv::Mat - the buffer is `int*`

Reply:
I got it! Fixed in last commit.